File: Easy/121_Best_Time_To_Buy_And_Sell_Stock/buy_and_sell.py
from typing import List
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        buy_day = 0
        sell_day = 1

        while sell_day < len(prices):
            if prices[buy_day] < prices[sell_day] :
                max_profit = max(max_profit, prices[sell_day] - prices[buy_day])
            else:
                buy_day = sell_day
            sell_day += 1
        return max_profit

File: Easy/121_Best_Time_To_Buy_And_Sell_Stock/buy_and_sell.go
package besttimetobuyandsellstock

func maxProfit(prices []int) int {

	maxProfit := 0
	buyDay := 0
	sellDay := 1

	for sellDay < len(prices) {
		if prices[buyDay] < prices[sellDay] {
			maxProfit = max(maxProfit, prices[sellDay]-prices[buyDay])
		} else {
			buyDay = sellDay
		}
		sellDay++
	}

	return maxProfit

}
